aiven/client/pretty.py

```diff
@@ -124,7 +124,7 @@ def iter_values(key, value):
         for key, value in item.items():
             if key in drop_fields:
                 continue  # field will not be printed
-            if table_layout is not None and key not in flattened_table_layout:
+            if table_layout is not None and not any(tcol.startswith(key) for tcol in flattened_table_layout):
                 continue  # table_layout has been specified but this field will not be printed
             for subkey, subvalue in iter_values(key, value):
```

Comment:
```suggestion
            for subkey, subvalue in iter_values(key, value):
                 if table_layout is not None and subkey not in flattened_table_layout:
                    continue  # table_layout has been specified but this field will not be printed
```
Your way won't work in rare occasions like:
1. There are fields `a` and `ab` in `table_layout`.
2. There are fields `a.b` and `a.c` in result, but only `a.b` is present in `table_layout`.

Reply:
Also, please consider writing a unit test which checks `yield_table` fix.